assets/src/edit-story/components/richText/editor.js

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { Editor } from 'draft-js';
+import PropTypes from 'prop-types';
+import {
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useImperativeHandle,
+  forwardRef,
+} from 'react';
+
+/**
+ * Internal dependencies
+ */
+import useUnmount from '../../utils/useUnmount';
+import useRichText from './useRichText';
+import customInlineDisplay from './customInlineDisplay';
+
+function RichTextEditor({ content, onChange }, ref) {
+  const editorRef = useRef(null);
+  const {
+    state: { editorState },
+    actions: {
+      setStateFromContent,
+      updateEditorState,
+      getHandleKeyCommand,
+      getContentFromState,
+      clearState,
+    },
+  } = useRichText();
+
+  // Load state from parent when content changes
+  useEffect(() => {
+    setStateFromContent(content);
+  }, [setStateFromContent, content]);
+
+  // Push updates to parent when state changes
+  useEffect(() => {
+    const newContent = getContentFromState(editorState);
+    if (newContent) {
```

Comment:
Hm, good question. Can you clear a text field and if so, what should happen? Currently it'll revert to what it was before entering edit mode because of this exact line.

Reply:
I think this is okay for now. We can look at it later if need be.

Reply:
When text is cleared, we should just display placeholder again. Ok to defer.